src/main/msp/msp.c

```diff
@@ -3138,28 +3289,27 @@ static mspResult_e mspProcessInCommand(mspDescriptor_t srcDesc, int16_t cmdMSP,
 #ifdef USE_SIMPLIFIED_TUNING
     // Added in MSP API 1.44
     case MSP_SET_SIMPLIFIED_TUNING:
-        currentPidProfile->simplified_pids_mode = sbufReadU8(src);
-        currentPidProfile->simplified_master_multiplier = sbufReadU8(src);
-        currentPidProfile->simplified_roll_pitch_ratio = sbufReadU8(src);
-        currentPidProfile->simplified_i_gain = sbufReadU8(src);
-        currentPidProfile->simplified_d_gain = sbufReadU8(src);
-        currentPidProfile->simplified_pi_gain = sbufReadU8(src);
-#ifdef USE_D_MIN
-        currentPidProfile->simplified_dmin_ratio = sbufReadU8(src);
-#else
-        sbufReadU8(src);
-#endif
-        currentPidProfile->simplified_feedforward_gain = sbufReadU8(src);
-        currentPidProfile->simplified_pitch_pi_gain = sbufReadU8(src);
-
-        currentPidProfile->simplified_dterm_filter = sbufReadU8(src);
-        currentPidProfile->simplified_dterm_filter_multiplier = sbufReadU8(src);
-
-        gyroConfigMutable()->simplified_gyro_filter = sbufReadU8(src);
-        gyroConfigMutable()->simplified_gyro_filter_multiplier = sbufReadU8(src);
-
-        applySimplifiedTuning(currentPidProfile);
-
+        {
+            bool apply = sbufReadU8(src);
+            readSimplifiedPids(currentPidProfile, src);
+            readSimplifiedDtermFilters(currentPidProfile, src);
+            readSimplifiedGyroFilters(gyroConfigMutable(), src);
+            if (apply) {
+                applySimplifiedTuning(currentPidProfile, gyroConfigMutable());
+            }
```

Comment:
Just noticed that ```readSimplifiedGyroFilters()``` and ```readSimplifiedGyroFilters()``` contain actual filter parameters in addition to the simplified tuning parameters. This means that it's already broken. We can't just request ```MSP_SIMPLIFIED_TUNING``` and then write the modified tuning parameters back using ```MSP_SET_SIMPLIFIED_TUNING``` anymore. The bool at the beginning also shifts everything in the message by 1 byte.

Reply:
The bool is needed so that we don't have to make a separate MSP message for just setting values.
Luckily we did not release 4.3 so we don't have to worry about backwards compatibility between different 4.3 versions? COnfigurator PR has already been adjusted.

Unfortunately or luckily to recalculate gyro and dterm sliders we have to know the values of the filters. So sliders values just by themselves are not enough. That's the consequence of the decision that filter sliders don't adjust lpf's that are zero, if you check simplified_tuning.c

Reply:
The way it works for these messages where we have get and set messages is that we can request the data, modify it and write it back in the same order using the SET message. This breaks that pattern so we can't do that anymore.

Reply:
Should moving the bool to the end of the MSP message be sufficient?

Reply:
@limonspb can't you refactor so the additional bytes needed are only used in the new MSP commands, not in the existing ones, as these should be symmetric?

Reply:
@haslinghuis or reserve a byte for it by sending a dummy byte at the start of ```MSP_SIMPLIFIED_TUNING```. This way we will avoid having it in the middle of the message or being shifted to a new index at the end if new parameters are added.  Still the positions of each byte will be wrong in ```MSP_SET_SIMPLIFIED_TUNING``` because of the added filter parameters that are not part of ```MSP_SIMPLIFIED_TUNING```. 

When these messages are different we can't use them in the lua scripts anymore without doing an additional ```MSP_FILTER_CONFIG``` request for the simplified tuning page. This eats up both memory and bandwidth.

Have you guys considered just sending ```MSP_SET_FILTER_CONFIG``` first followed by ```MSP_SET_SIMPLIFIED_TUNING```?

Reply:
Yeah I was doing something like that in my first commit of the configurator PR implementing this change change (in commit 2). But the problem with that it is making changes to firmware without the user has hit the save button.

Yes I think these extra fields should be part of the new MSP commands, these are used to set / get values without saving to firmware anyway. The existing commands should be left alone.

Reply:
@klutvott123  @haslinghuis  i can add reserved bytes and make commands symmetric.
But that's like extra code increasing the size of the FLASH memory, isn't it?

But if you think I should add them, please let's have a list which and where:
1. extra dummy byte at the beginning of "MSP_SIMPLIFIED_TUNING"
what else?

Reply:
I don't think adding reserved bytes in the middle of a message is a good solution. How many do we need? What if we run out?
New parameters should always be added to the end of a message. Messages should also be as specific as possible and a message called SIMPLIFIED_TUNING should contain exactly that. Simplified tuning parameters. Honestly I think these messages should stay as they are.

> I have no clue how LUA works,

When you open the simplified tuning page in the lua scripts they will do a request for ```MSP_SIMPLIFIED_TUNING``` and present those values on screen. The modified values will only be sent back using ```MSP_SET_SIMPLIFIED_TUNING``` when the user selects "save page" from the menu. There's no communication with the FC unlesse the user presses a button to initiate it. This works perfectly in the scripts. Each page in the lua scripts is only allowed to do ONE request as doing multiple requests will just take too long.

Reply:
Instead of 
```
case MSP_SET_SIMPLIFIED_TUNING:
        {
            bool apply = sbufReadU8(src);
            readSimplifiedPids(currentPidProfile, src);
            readSimplifiedDtermFilters(currentPidProfile, src);
            readSimplifiedGyroFilters(gyroConfigMutable(), src);
            if (apply) {
                applySimplifiedTuning(currentPidProfile, gyroConfigMutable());
            }
        }
        break;
```
We should use the validate bytes using MSP_VALIDATE_SIMPLIFIED_TUNING so we don't need to store it here also.

Reply:
@haslinghuis this is not storing apply byte. This is a byte that Configurator/LUA sends to FW and tells if FW should recalculate values or not.
@klutvott123  i understand if we are talking about MSP commands that are already been released. But for the new commands in 4.3 we should be free to change them as we need. We do not need to support backwards compatibility between different 4.3 versions.

Reply:
@limonspb You're right tha that we don't need to keep compatibility between different 4.3 versions. 
The issue here is that new parameters will have to be added to the middle of a message in the future. So how many reserved bytes do we need? What if we run out? Then we have to break compatibility or start adding new parameters to the end of the message like it should have been to begin with.

Reply:
Just had a chat with @limonspb about this and whilst in general I think it's not a good idea to pad out commands with spare bytes just in case we need them in future, in this case there is good reason to do so. The extra flash code space required to read/write such extra bytes will be much smaller that the code that would be required to support additional expanded, or parameterised commands in a future release. This is particularly the case where it's not straightforward to just tag bytes onto the end of a command, but rather they need to be inserted mid-command.